instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py

```diff
@@ -115,12 +134,35 @@ class BaseTest(TestBase, metaclass=abc.ABCMeta):
         # pylint: disable=invalid-name
         def setUp(self):
             super().setUp()
+            test_name = ""
+            if hasattr(self, "_testMethodName"):
+                test_name = self._testMethodName
+            sem_conv_mode = "default"
+            if "new_semconv" in test_name:
+                sem_conv_mode = "http"
+            elif "both_semconv" in test_name:
+                sem_conv_mode = "http/dup"
+            self.env_patch = mock.patch.dict(
+                "os.environ",
+                {
+                    OTEL_SEMCONV_STABILITY_OPT_IN: sem_conv_mode,
+                },
+            )
+            self.env_patch.start()
```

Comment:
Wondering if instead of making this under the hood we can make it more explicit and add the three decorators instead, e.g `@default_http_semconv`, `@new_http_semconv`, `@both_http_semconv`

Maybe we can avoid the one for the default one if the others can override the patch started here on setUp.

Reply:
I'm happy to do that in a follow-up PR if you're fine with that. My concern is about these changes being compatible with what we already have in other transitions; after this gets merged, I can work on a better approach. 
@lzchen let me know if you are you ok with that as well.

Reply:
Sure